sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokers.java

```diff
@@ -159,7 +200,27 @@ public String subclass(TypeDescription.Generic superClass) {
             .method(ElementMatchers.named("invokeSetup"))
             .intercept(delegateOrNoop(signature.setup()))
             .method(ElementMatchers.named("invokeTeardown"))
-            .intercept(delegateOrNoop(signature.teardown()));
+            .intercept(delegateOrNoop(signature.teardown()))
+            .method(ElementMatchers.named("invokeGetInitialRestriction"))
+            .intercept(delegateWithDowncastOrThrow(signature.getInitialRestriction()))
+            .method(ElementMatchers.named("invokeSplitRestriction"))
+            .intercept(
+                (signature.splitRestriction() == null)
+                    ? MethodDelegation.to(DefaultSplitRestriction.class)
+                    : new DowncastingParametersMethodDelegation(
+                        signature.splitRestriction().targetMethod()))
+            .method(ElementMatchers.named("invokeGetRestrictionCoder"))
+            .intercept(
+                signature.processElement().isSplittable()
+                    ? ((signature.getRestrictionCoder() == null)
```

Comment:
Nested ternary conditionals is the limit -- we should add a method for each of these interceptions. Something like:
`delegateSplitOrDefault(...)`
`delegateProcessElement(...)`

and then use normal if statements.